release-notes/3.0/preview/api-diff/preview8/.Net/3.0-preview8_System.Text.Json.md

```diff
@@ -0,0 +1,53 @@
+# System.Text.Json
+
+``` diff
+ namespace System.Text.Json {
+     public sealed class JsonDocument : IDisposable {
++        public void WriteTo(Utf8JsonWriter writer);
+     }
+     public readonly struct JsonElement {
+-        public void WriteProperty(ReadOnlySpan<byte> utf8PropertyName, Utf8JsonWriter writer);
+
+-        public void WriteProperty(ReadOnlySpan<char> propertyName, Utf8JsonWriter writer);
+
+-        public void WriteProperty(string propertyName, Utf8JsonWriter writer);
+
+-        public void WriteProperty(JsonEncodedText propertyName, Utf8JsonWriter writer);
+
++        public void WriteTo(Utf8JsonWriter writer);
+-        public void WriteValue(Utf8JsonWriter writer);
+
+     }
+     public readonly struct JsonEncodedText : IEquatable<JsonEncodedText> {
+-        public static JsonEncodedText Encode(ReadOnlySpan<byte> utf8Value);
+
++        public static JsonEncodedText Encode(ReadOnlySpan<byte> utf8Value, JavaScriptEncoder encoder = null);
+-        public static JsonEncodedText Encode(ReadOnlySpan<char> value);
+
++        public static JsonEncodedText Encode(ReadOnlySpan<char> value, JavaScriptEncoder encoder = null);
+-        public static JsonEncodedText Encode(string value);
+
++        public static JsonEncodedText Encode(string value, JavaScriptEncoder encoder = null);
+     }
+     public readonly struct JsonProperty {
++        public void WriteTo(Utf8JsonWriter writer);
+     }
+     public static class JsonSerializer {
+-        public static string Serialize(object value, Type inputType, JsonSerializerOptions options = null);
++        public static string Serialize(object value, Type type, JsonSerializerOptions options = null);
+-        public static void Serialize(Utf8JsonWriter writer, object value, Type inputType, JsonSerializerOptions options = null);
++        public static void Serialize(Utf8JsonWriter writer, object value, Type type, JsonSerializerOptions options = null);
+-        public static Task SerializeAsync(Stream utf8Json, object value, Type inputType, JsonSerializerOptions options = null, CancellationToken cancellationToken = default(CancellationToken));
++        public static Task SerializeAsync(Stream utf8Json, object value, Type type, JsonSerializerOptions options = null, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
I don't recall discussing renaming the Type parameter name. Looks like it was unintentionally changed here: https://github.com/dotnet/corefx/pull/39524, which reverted https://github.com/dotnet/corefx/pull/38933

The Deserialize method have the parameter named "returnType"

cc @steveharter, @terrajobst